webApp/server_modules/routes/profile.js

```diff
@@ -138,7 +117,35 @@ router.post('/user/:id(\\d+)/review/:course(\\d+)/sub', (req, res) => {
 
 
 // A route that will actually create a user's account within the database
+router.post('/createProfile', function(req, res) {
+    let profile = newProfile(req.body, req.user.id, req.user.extra);
+    addUser(profile)
+        .then(profile => {
+            req.login({ id: profile.googleID }, err => {
+                if (err) return res.redirect('/');
+                res.redirect('/');
+            });
```

Comment:
Everything after this line might cause an error due to `res.redirect()` being before it all . A possible fix would be moving all tutor adding functionality before `req.login()`
```suggestion
'insert the tutor adding functionality here once done then do the rest'
    req.login(....
        res.redirect('/');
            });
```